osquery/tables/yara/yara.cpp

```diff
@@ -97,6 +101,44 @@ static YARAConfigParser getYaraParser(void) {
   return yaraParser;
 }
 
+Status processRequest(const std::string& url_name, std::string& rules) {
+  auto yaraParser = getYaraParser();
+  if (isNull(yaraParser)) {
+    return Status::failure("YARA config parser plugin not found");
+  }
+
+  const std::string url = yaraParser->getURL(url_name);
+  if (url.empty()) {
+    return Status::failure("signature url not valid");
+  }
+
+  try {
+    http::Client client(TLSTransport().getOptions());
+    http::Response response;
+    http::Request request(url);
+
+    response = client.get(request);
+    auto mod_time = response.headers()["Last-Modified"];
```

Comment:
If I read this correctly, and I might not, this does not look like a correct implementation of http caching. I think this is doing a complete fetch, then checking if the `Last-Modified` header has changed. As you've already _done_ the fetch, there isn't much point to the cache.

I think caching has a couple of approaches, and ideally this is something some underlying library could implement. They might use `etag` or they might use `last-modified`. They might be conditional headers (like `If-Modified-Since` or `If-Match`), or they might do a `HEAD` to see if a subsequent `GET` is even needed. And probably, they'd use all of the above.

I don't know if it's better to push caching off to future work. Possible in the http client itself.

Reply:
Thanks @directionless  for the comments. I will have a second look into the caching mechanism and will push it to the future work.